src/overlays/actors/ovl_En_Go/z_en_go.c

```diff
@@ -567,205 +853,236 @@ s32 func_80A122EC(EnGo* this) {
-
     CollisionCheck_SetOC(play, &play->colChkCtx, &this->colliderSphere.base);
 }
 
-void func_80A125BC(EnGo* this, PlayState* play) {
+/**
+ * Update the collider for the frozen form of a Goron.
+ */
+void EnGo_UpdateFrozenCollider(EnGo* this, PlayState* play) {
     s32 pad;
 
     Math_Vec3f_ToVec3s(&this->colliderCylinder.dim.pos, &this->actor.world.pos);
-
     this->colliderCylinder.dim.radius = 46;
     this->colliderCylinder.dim.height = 78;
-
-    if (this->unk_3C6 == 0) {
+    if (this->harmlessTimer == 0) {
         CollisionCheck_SetAT(play, &play->colChkCtx, &this->colliderCylinder.base);
     }
     CollisionCheck_SetAC(play, &play->colChkCtx, &this->colliderCylinder.base);
     CollisionCheck_SetOC(play, &play->colChkCtx, &this->colliderCylinder.base);
 }
 
-void func_80A12660(EnGo* this, PlayState* play) {
+/**
+ * Update the collider for all standing forms of a Goron.
+ */
+void EnGo_UpdateStandingCollider(EnGo* this, PlayState* play) {
     Math_Vec3f_ToVec3s(&this->colliderCylinder.dim.pos, &this->actor.world.pos);
     this->colliderCylinder.dim.radius = 24;
     this->colliderCylinder.dim.height = 62;
     CollisionCheck_SetOC(play, &play->colChkCtx, &this->colliderCylinder.base);
 }
 
-void func_80A126BC(EnGo* this, PlayState* play) {
+/**
+ * Update the collider for the Goron.
+ *
+ * Both the rolling snowball and frozen forms can hurt Player on collision, others do not.
+ */
+void EnGo_UpdateCollider(EnGo* this, PlayState* play) {
     Player* player = GET_PLAYER(play);
 
     if (!(player->stateFlags2 & PLAYER_STATE2_4000)) {
-        if (this->unk_3C6 != 0) {
-            this->unk_3C6--;
-        }
-    }
-
-    if (ENGO_GET_F(&this->actor) == ENGO_F_8) {
-        func_80A124A0(this, play);
-    } else if (this->unk_390 & 0x100) {
-        func_80A123A0(this, play);
-    } else if (this->unk_390 & 0x200) {
-        func_80A124FC(this, play);
-    } else if (this->unk_390 & 0x400) {
-        func_80A125BC(this, play);
+        DECR(this->harmlessTimer);
+    }
+    if (ENGO_GET_TYPE(&this->actor) == ENGO_MEDIGORON) {
+        EnGo_UpdateMedigoronCollider(this, play);
+    } else if (this->actionFlags & ENGO_FLAG_SNOWBALLED) {
+        EnGo_UpdateSnowballCollider(this, play);
+    } else if (this->actionFlags & ENGO_FLAG_ROLLED_UP) {
+        EnGo_UpdateRolledUpCollider(this, play);
+    } else if (this->actionFlags & ENGO_FLAG_FROZEN) {
+        EnGo_UpdateFrozenCollider(this, play);
     } else {
-        func_80A12660(this, play);
+        EnGo_UpdateStandingCollider(this, play);
     }
 }
 
-s32 func_80A12774(EnGo* this, PlayState* play) {
-    if (!(this->unk_390 & 7) || !Actor_ProcessTalkRequest(&this->actor, &play->state)) {
+/**
+ * Check if the actor is talking, and if so set the action function to handle it.
+ *
+ * @return True if talking
+ */
+s32 EnGo_UpdateTalking(EnGo* this, PlayState* play) {
+    if (!(this->actionFlags & 7) || !Actor_ProcessTalkRequest(&this->actor, &play->state)) {
```

Comment:
There are a handfull of calls to SubsS_UpdateFlags with setBits of 0, 3, and 4, and clearBits of 7. This pattern appears in the stateFlags/actionFlags of many other actors, and it doesn't appear anyone has named them yet (with the exception of EN_NB_FLAG_<#>). Since it appeared to be a common pattern, and I was unable to figure out good names to use, I left the literals in place, assuming it would be easier to identify later when good names, likely non-specific to the goron actor are chosen.

I could do EN_GO_FLAG_<#> following the NB example, but I think that might just obfuscate that these flags appear to be common use.